bookings/views.py

```diff
@@ -1,14 +1,15 @@
 """ Views for the bookings app. """
 import datetime
-from django.shortcuts import render, redirect, reverse, get_object_or_404
+
 from django.contrib import messages
 from django.contrib.auth.decorators import login_required
+from django.shortcuts import render, redirect, reverse, get_object_or_404
 
 from restaurant.models import Restaurant
-from .models import Booking
-from .forms import BookingForm
 from .check_availability import create_booking_slots
 from .confirmation_email import send_confirmation_email
+from .forms import BookingForm
+from .models import Booking
 
 
 def make_booking(request):
```

Comment:
When you get heavily nested logic, it creates a lot of potential paths through the code which in turn makes testing and understanding more complex.

Considering how to reduce nesting can be tricky, but not impossible. Here for example, if you flip the logic of the first if statement, you can set the form and context and return early (request.method != 'POST'). Having lots of return statements can also be considered a code smell, but you are then looking at a redesign or accepting that some times it is the right thing to do.

Edit: Same deal with 'update_booking)